src/api/utils/resourceHookFactories.ts

```diff
@@ -19,14 +20,24 @@ export const createUseQuery = <Result>(
   key: string[],
   url: string,
   fetchOptions?: RequestInit
-): (() => UseQueryResult<Result>) => {
+): ((
+  options?: Omit<
+    UseQueryOptions<unknown, unknown, Result, string[]>,
+    'queryKey' | 'queryFn'
```

Comment:
beautiful. one thing is that since you are using the `UseQueryOptions` with the same values in two places, you could pull that out in a type within this file. Just to remove the repeated inline types. I still approve tho, looks great.

Reply:
I had to rerun the tests anyway because of flakiness, so I made the change to trigger a rerun. Have a look. :)